--- a/src/libFLAC/include/private/cpu.h	2015-12-22 06:12:20.370104292 +0100
+++ b/src/libFLAC/include/private/cpu.h	2015-12-22 06:13:00.362103123 +0100
@@ -72,16 +72,6 @@
     #define FLAC__FMA_SUPPORTED 1
   #endif
 #elif defined __GNUC__
-  #if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 9)) /* since GCC 4.9 -msse.. compiler options aren't necessary */
-    #define FLAC__SSE_TARGET(x) __attribute__ ((__target__ (x)))
-    #define FLAC__SSE_SUPPORTED 1
-    #define FLAC__SSE2_SUPPORTED 1
-    #define FLAC__SSSE3_SUPPORTED 1
-    #define FLAC__SSE4_1_SUPPORTED 1
-    #define FLAC__AVX_SUPPORTED 1
-    #define FLAC__AVX2_SUPPORTED 1
-    #define FLAC__FMA_SUPPORTED 1
-  #else /* for GCC older than 4.9 */
     #define FLAC__SSE_TARGET(x)
     #ifdef __SSE__
       #define FLAC__SSE_SUPPORTED 1
@@ -104,7 +94,6 @@
     #ifdef __FMA__
       #define FLAC__FMA_SUPPORTED 1
     #endif
-  #endif /* GCC version */
 #endif /* compiler version */
 #endif /* intrinsics support */
 
