--- universalindentgui-1.2.0-r1.ebuild	2015-02-28 01:24:08.625853944 +0100
+++ files/universalindentgui-1.2.0-r1.ebuild	2015-02-28 01:23:09.817854480 +0100
@@ -15,7 +15,7 @@
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="debug examples html perl php python ruby xml"
+IUSE="debug examples html perl php python ruby xml uncrustify"
 
 DEPEND="dev-qt/qtcore:4
 	dev-qt/qtgui:4
@@ -34,6 +34,7 @@
 	php? ( dev-php/PEAR-PHP_Beautifier )
 	ruby? ( dev-lang/ruby )
 	xml? ( dev-util/xmlindent )
+	uncrustify? ( dev-util/uncrustify )
 "
 
 DOCS="CHANGELOG.txt readme.html"
@@ -99,6 +100,10 @@
 		UIGUIFILES="${UIGUIFILES} xmlindent"
 	fi
 
+	if use uncrustify; then
+		UIGUIFILES="${UIGUIFILES} uncrustify"
+	fi
+
 	local IFILES= I=
 	for I in ${UINDENTERS}; do
 		IFILES="${IFILES} indenters/${I}"
